Fix #85

fbuetler opened the pull request:
Fixes https://github.com/eclipse-vertx/vertx-http-proxy/issues/85

`null` is used as a signal, so we should not set a default in the constructor. 

https://github.com/fbuetler/vertx-http-proxy/blob/a97630bc9ee465f073fbefd5febbe12cdc43d78b/src/main/java/io/vertx/httpproxy/impl/ProxiedRequest.java#L173

With this, the updated reproducer provided in the issue passes.

vietj commented on src/main/java/io/vertx/httpproxy/impl/ProxiedRequest.java:
```diff
@@ -72,7 +72,6 @@ public ProxiedRequest(HttpServerRequest proxiedRequest) {
     this.absoluteURI = proxiedRequest.absoluteURI();
     this.proxiedRequest = proxiedRequest;
     this.context = (ContextInternal) ((HttpServerRequestInternal) proxiedRequest).context();
-    this.authority = proxiedRequest.authority();
```
I'd prefer this.authority = null with a comment

fbuetler commented:
There was a test that should have caught this issue but was implemented incorrectly. 

The test was setting the `host` header on the request that is created in the proxy, but in the proxy implementation [the host is ignored](https://github.com/eclipse-vertx/vertx-http-proxy/blob/78a73651a5ddd71674a1ae0ae12c0b4d50d22ad9/src/main/java/io/vertx/httpproxy/impl/ProxiedRequest.java#L172), unfortunately.

I fixed the test and could verify that without the fix included in this PR, the test would fail but if included, it succeeds.

I also fixed some whitespace issues, if you don't mind including them here.

tsegismont reviewed (approved):
LGTM, thank you @fbuetler

tsegismont commented:
Can you please look at the test failures? Thanks

fbuetler commented:
Ready for another round. They pass now on my machine.

tsegismont reviewed (approved):
Thanks for the udates @fbuetler !
